sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -40,6 +40,14 @@ sealed abstract class Filter {
     case f: Filter => f.references
     case _ => Array.empty
   }
+
+  /**
+   * If any of the references of this filter contains nested field
+   * @since 3.2.0
+   */
+  def containNestedField: Boolean = {
+    this.references.exists(_.contains("."))
```

Comment:
What if there is a column name containing `.`

Reply:
How about pass the schema or top-level output attributes as a parameter, and judge if the filter is of a nested column.

Reply:
This is a quick implementation. I plan to write a parser to parse nested column name. 

For example, `a.b` is a nested field while
```
`a.b`
```
is a field name containing `dot`. WDYT?

Reply:
SGTM